paywall/src/services/walletService.js

```diff
@@ -56,8 +56,8 @@ export default class WalletService extends EventEmitter {
    * @return
    */
   async connect(providerName) {
-    if (providerName === this.providerName) {
-      // If the provider did not really change, no need to reset it
+    if (providerName && providerName === this.providerName) {
```

Comment:
Any reason why this change is in that PR? Would you mind putting it in another one otherwise?